src/main/java/net/sf/jabref/external/push/AbstractPushToApplication.java

```diff
@@ -32,23 +33,19 @@
 /**
  * Class for pushing entries into TexMaker.
  */
-public class PushToTexmaker implements PushToApplication {
-
-    private boolean couldNotCall;
-    private boolean notDefined;
-    private JPanel settings;
-    private final JTextField texmakerPath = new JTextField(30);
-    private final JTextField citeCommand = new JTextField(30);
+public abstract class AbstractPushToApplication implements PushToApplication {
```

Comment:
Trade-off between supporting special cases in the abstract class and keeping almost identical code in the different PushTo classes, it should have read.